php/commands/site.php

```diff
@@ -161,15 +162,22 @@ public function set_connection_mode($args, $assoc_args) {
   if (in_array($env->get('id'), array('test', 'live'))) {
     $this->failure('Connection mode cannot be set in Test or Live environments');
   }
-  if ($mode == $env->info('connection_mode')) {
-    $this->failure(
-      'The connection mode on {site} for {env} is already set to {mode}.',
-      array(
-        'site' => $site->get('name'),
-        'env' => $env->get('id'),
-        'mode' => $mode
-      ),
-      -1
+  try {
+    $current_mode = $env->info('connection_mode');
+    if ($current_mode == $env->info('connection_mode')) {
+      $this->failure(
+        'The connection mode on {site} for {env} is already set to {mode}.',
+        array(
+          'site' => $site->get('name'),
+          'env' => $env->get('id'),
+          'mode' => $mode
+        ),
+        -1
+      );
+    }
+  } catch (TerminusException $e) {
```

Comment:
I approved this change already, but in the future we might think about how to document exception catching so we know that the point of this is to catch the situation in which the connection mode is unavailable (not a failure of the failure logger, etc.). Like maybe wrap the connection-mode check in a method inside the Site model, and handle the exception there.